syntax="proto3";

package ServerRpc.msg;

message msg {
    string fromUserID  = 1;
    string toUserID = 2;
    string groupID = 8;
    string content = 3;
    int64 createTime = 4;
    int64 seq = 5;
    int32 platformID=6;
    int32 msgType=7;
}

message sendMsgReq {
    msg msg_data = 1;
}

message sendMsgResp {
    int64 sendTime = 1;
}

message pullMsgs {
    repeated msg msgs = 1;
}

message setHasReadSeqReq {
    string fromUserID = 1;
    string toUserID = 2;
    string groupID = 3;
    int32 msgType =4;
    int64 seq = 5;
}

message setHasReadSeqResp {
    int64 sendTime = 1;
}

message getHasReadSeqAndMaxSeqReq {
    string fromUserID = 1;
    string toUserID = 2;
    string groupID = 3;
    int32 msgType =4;
}

message getHasReadSeqAndMaxSeqResp {
    int64 hasReadSeq = 1;
    int64 maxSeq = 2;
}

message syncMsgsReq {
    string fromUserID = 1;
    string toUserID = 2;
    string groupID = 3;
    int32 msgType =4;
    int64 startSeq = 5;
    int64 endSeq = 6;
}

message syncMsgsResp {
    pullMsgs msgs = 1;
}

service Msg {
    rpc sendMsg(sendMsgReq) returns (sendMsgResp);
    rpc syncMsgs(syncMsgsReq) returns (syncMsgsResp);
    rpc setHasReadSeq(setHasReadSeqReq) returns (setHasReadSeqResp);
    rpc getHasReadSeqAndMaxSeq(getHasReadSeqAndMaxSeqReq) returns (getHasReadSeqAndMaxSeqResp);
}